core/field.js

```diff
@@ -603,8 +603,12 @@ Blockly.Field.prototype.getDisplayText_ = function() {
   // Replace whitespace with non-breaking spaces so the text doesn't collapse.
   text = text.replace(/\s/g, Blockly.Field.NBSP);
   if (this.sourceBlock_.RTL) {
-    // The SVG is LTR, force text to be RTL.
-    text += '\u200F';
+    // The SVG is LTR, force text to be RTL unless a number.
+    if (this.sourceBlock_.editable_ && this.sourceBlock_.type === 'math_number') {
+      text += '\u200E';
+    } else {
+      text = '\u202B' + text + '\u202C';
```

Comment:
Can have negative numbers with the '-' on the right (`10-`) by just using line 610.

Reply:
We have a big problem with the blocks!
It seems that {direction:rtl} sould be use only on some blocks. On other blocks it causes things like:
![capture](https://user-images.githubusercontent.com/1233716/44848803-c95db680-ac60-11e8-96de-6fdb720e3673.PNG)

And it also causes stuff like the "10-" instead of "-10".

Note that all math notation and expression should be purely LTR. 
Maybe the solution is not to use RTL at all in the client, and revise the translation/json so that it will be rendered correctly is the client. 
Another option is to mark expression in the translation that should be rendered RTL (maybe by some prefix), and the rest will be LTR  by default.

Reply:
@guyhed in this version most things are better - for example, < and > give the expected results. Subtraction and division can be handled in the translation by switching the order of the arguments in the translation knowing that they will be rendered RTL.